src/api/dotnet/Context.cs

```diff
@@ -956,17 +985,15 @@ public BoolExpr MkXor(IEnumerable<BoolExpr> ts)
             Debug.Assert(ts != null);
             Debug.Assert(ts.All(a => a != null));
             CheckContextMatch<BoolExpr>(ts);
-            BoolExpr r = null;
-            foreach (var t in ts)
-            {
-                if (r == null)
-                    r = t;
-                else
-                    r = MkXor(r, t);
-            }
-            if (r == null)
-                r = MkTrue();
-            return r;
+
+            return ts.Any()
```

Comment:
This was probably the trickiest one to get right. I believe what the original code was doing is essentially a fold / aggregate (although the empty case was returning `MkTrue()`, which isn't the initial value for Xor, but I've preserved that behaviour here).

The trickiness with the disposing is that we only want to dispose of the temporary objects created as part of the fold. To ensure this is the case I seeded `Aggregate` with `MkFalse` so that in the first call to the lambda it will dispose of this temp object once the `MkXor` has been called. Each subsequent call to the lambda will dispose of `MkXor` created in the previous invocation. Therefore only objects explicitly created in this method (the intermediate values of the fold) are passed to the `using` statement. The final value is not disposed of as it is passed back to the caller and therefore they own it.

In the trivial case of an empty list therefore is nothing to dispose because we just return `MkTrue()`.